### ISSUE TITLE
cpptools doesn't have execute permission

### ISSUE THREAD

**Issue by PaulPrice** [microsoft/vscode-cpptools#9265]
Issue Type: <b>Bug</b>

I am running VSCode on MacOS using a devcontainer. On installing the C/C++ extension, I get output:
```
[Error - 9:29:51 PM] Starting client failed
Launching server using command /home/pfs/.vscode-server/extensions/ms-vscode.cpptools-1.9.8-linux-x64/bin/cpptools failed.
```
That file does not have the user-execute bit set:
```
-rw-r--r-- 1 pfs pfs 21191216 May  2 21:29 /home/pfs/.vscode-server/extensions/ms-vscode.cpptools-1.9.8-linux-x64/bin/cpptools
```
Setting the user-execute bit manually allows `cpptools` to work fine.

I'm not sure why the user-execute bit would not be set for me on install. I'm happy to provide copies of my Docker image and `devcontainer.json` and `devcontainer.env` if it would help.

Extension version: 1.9.8
VS Code version: Code 1.66.2 (dfd34e8260c270da74b5c2d86d61aee4b6d56977, 2022-04-11T07:49:20.994Z)
OS version: Darwin x64 19.6.0
Restricted Mode: No
Remote OS version: Linux x64 5.10.104-linuxkit
Remote OS version: Linux x64 5.10.104-linuxkit
Remote OS version: Linux x64 3.10.0-1160.62.1.el7.x86_64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|4, 4, 4|
|Memory (System)|16.00GB (0.15GB free)|
|Process Argv|-psn_0_25729160 --crash-reporter-id 76307722-6a57-4297-8ec9-61983f634a11|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|Dev Container: pipe2d|
|OS|Linux x64 5.10.104-linuxkit|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (4 x 2600)|
|Memory (System)|7.77GB (2.93GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|Dev Container: lsst|
|OS|Linux x64 5.10.104-linuxkit|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (4 x 2600)|
|Memory (System)|7.77GB (2.93GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: tiger|
|OS|Linux x64 3.10.0-1160.62.1.el7.x86_64|
|CPUs|Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (40 x 2543)|
|Memory (System)|187.20GB (58.77GB free)|
|VM|0%|
</details>

**Comment by Colengms** [comment]
Hi @PaulPrice .   The C/C++ extension will attempt to add execute permissions to its native binaries when it's activated.  If you are still able to repro the issue, could you enable `"C_Cpp.loggingLevel": "Debug"` and provide the log output of the C/C++ channel leading up to the issue?  Is it possible the environment/account you have configured does not provide VS Code with sufficient privileges for its child processes to successfully execute a `chmod`?  The relevant code is in our TypeScript component, which can be referenced [here](https://github.com/microsoft/vscode-cpptools/blob/85c9b30c05631b7c6ba501d0a1018f8b1b9a6e6b/Extension/src/common.ts#L744-L786) and [here](https://github.com/microsoft/vscode-cpptools/blob/85c9b30c05631b7c6ba501d0a1018f8b1b9a6e6b/Extension/src/main.ts#L134-L163).

**Comment by PaulPrice** [comment]
* Deleted `.vscode-server/extensions/ms-vscode.cpptools-*` and `.vscode-server/extensionsCache/ms-vscode.cpptools-*`
* Set `C_Cpp.loggingLevel` to `Debug`
* Rebuild container
* Extensions -> C/C++ -> Install in dev container -> "Launching server using command ... failed"
* There doesn't appear to be any "C/C++" channel in the "OUTPUT" tab dropdown, only "cpptools", which contains only:
```
[Error - 10:55:05 PM] Starting client failed
Launching server using command /home/pfs/.vscode-server/extensions/ms-vscode.cpptools-1.9.8-linux-x64/bin/cpptools failed.
```
* Terminal -> `chmod u+x /home/pfs/.vscode-server/extensions/ms-vscode.cpptools-1.9.8-linux-x64/bin/cpptools`
* Reload window
* Now there's a "C/C++" channel in the "OUTPUT" dropdown, and it contains:
```
loggingLevel: Debug
cpptools/didChangeCppProperties
Attempting to get defaults from C++ compiler in "compilerPath" property: '/opt/pfs/conda/current/envs/lsst-scipipe-3.0.0/bin/gcc'
Querying compiler for default C++ language standard using command line: "/opt/pfs/conda/current/envs/lsst-scipipe-3.0.0/bin/gcc" -x c++ -E -dM /dev/null
terminating child process: 7159
Detected language standard version: gnu++14
Querying compiler for default C language standard using command line: "/opt/pfs/conda/current/envs/lsst-scipipe-3.0.0/bin/gcc" -x c -E -dM /dev/null
terminating child process: 7161
Detected language standard version: gnu17
Querying compiler's default target using command line: "/opt/pfs/conda/current/envs/lsst-scipipe-3.0.0/bin/gcc" -dumpmachine
Compiler returned default target value: x86_64-conda-linux-gnu

terminating child process: 7164
Compiler query command line: "/opt/pfs/conda/current/envs/lsst-scipipe-3.0.0/bin/gcc" -Wall -Wno-unknown-pragmas -Wno-unused-local-typedefs -std=c++14 -Wp,-v -E -dM -x c++ -m64  /dev/null
terminating child process: 7165
Attempting to get defaults from C compiler in "compilerPath" property: '/opt/pfs/conda/current/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/current/envs/lsst-scipipe-3.0.0/bin/gcc" -Wall -Wno-unknown-pragmas -Wno-unused-local-typedefs -std=c11 -Wp,-v -E -dM -x c -m64  /dev/null
terminating child process: 7167
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include-fixed/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/sysroot/usr/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/afw/g31dc256b76+308b63831b/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/astshim/gb04e0da5b2+ce83317b7e/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/base/gc57f72a14a+ce83317b7e/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/cpputils/g1ffd6c4751+50c04d9c73/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/daf_base/gcc072456be+b6f9ac6643/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/eigen/g04a8d4365e/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/geom/g491c310abf+35ead245fa/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/log/gd27fc4addd+b6f9ac6643/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/meas_algorithms/gf36ae6ace1+1d7ae49801/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/meas_base/g0129b61b94+2131942eb1/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/pex_config/g9aff22451c+c55107240f/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/pex_exceptions/g958458c828+fc3c6d1a77/include/ will be indexed
  Folder: /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/sphgeom/g0f31348038+ce83317b7e/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/include/python3.8/ will be indexed
  Folder: /workspaces/pfs/drp_stella/ will be indexed
cpptools/didChangeCppProperties
Attempting to get defaults from C++ compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu++14 -Wp,-v -E -dM -x c++ -m64  /dev/null
terminating child process: 7170
Attempting to get defaults from C compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu17 -Wp,-v -E -dM -x c -m64  /dev/null
terminating child process: 7172
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include-fixed/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/sysroot/usr/include/ will be indexed
  Folder: /workspaces/pfs/pfs_pipe2d/ will be indexed
cpptools/didChangeCppProperties
Attempting to get defaults from C++ compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu++14 -Wp,-v -E -dM -x c++ -m64  /dev/null
terminating child process: 7175
Attempting to get defaults from C compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu17 -Wp,-v -E -dM -x c -m64  /dev/null
terminating child process: 7178
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include-fixed/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/sysroot/usr/include/ will be indexed
  Folder: /workspaces/pfs/datamodel/ will be indexed
cpptools/didChangeCppProperties
Attempting to get defaults from C++ compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu++14 -Wp,-v -E -dM -x c++ -m64  /dev/null
terminating child process: 7180
Attempting to get defaults from C compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu17 -Wp,-v -E -dM -x c -m64  /dev/null
terminating child process: 7182
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include-fixed/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/sysroot/usr/include/ will be indexed
  Folder: /workspaces/pfs/obs_pfs/ will be indexed
cpptools/didChangeCppProperties
Attempting to get defaults from C++ compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu++14 -Wp,-v -E -dM -x c++ -m64  /dev/null
terminating child process: 7184
Attempting to get defaults from C compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu17 -Wp,-v -E -dM -x c -m64  /dev/null
terminating child process: 7187
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include-fixed/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/sysroot/usr/include/ will be indexed
  Folder: /workspaces/pfs/drp_doc/ will be indexed
cpptools/didChangeCppProperties
Attempting to get defaults from C++ compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu++14 -Wp,-v -E -dM -x c++ -m64  /dev/null
terminating child process: 7189
Attempting to get defaults from C compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu17 -Wp,-v -E -dM -x c -m64  /dev/null
terminating child process: 7191
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include-fixed/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/sysroot/usr/include/ will be indexed
  Folder: /workspaces/pfs/drp_instmodel/ will be indexed
cpptools/didChangeCppProperties
Attempting to get defaults from C++ compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu++14 -Wp,-v -E -dM -x c++ -m64  /dev/null
terminating child process: 7193
Attempting to get defaults from C compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu17 -Wp,-v -E -dM -x c -m64  /dev/null
terminating child process: 7195
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include-fixed/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/sysroot/usr/include/ will be indexed
  Folder: /workspaces/pfs/spt_ExposureTimeCalculator/ will be indexed
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu17 -Wp,-v -E -dM -x c -m64  /dev/null
terminating child process: 7198
Code browsing service initialized
Attempting to get defaults from C++ compiler in "compilerPath" property: '/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc'
Compiler query command line: "/opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/bin/gcc" -std=gnu++14 -Wp,-v -E -dM -x c++ -m64  /dev/null
terminating child process: 7202
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include-fixed/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/include/ will be indexed
  Folder: /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/sysroot/usr/include/ will be indexed
  Folder: /workspaces/pfs/pfs_utils/ will be indexed
textDocument/didOpen: /workspaces/pfs/drp_stella/src/FiberTraceSet.cc
Discovering files...
cpptools/getDocumentSymbols: /workspaces/pfs/drp_stella/src/FiberTraceSet.cc (id: 2)
cpptools/textEditorSelectionChange
cpptools/activeDocumentChange: /workspaces/pfs/drp_stella/src/FiberTraceSet.cc
cpptools/getDocumentSymbols
  Processing folder (recursive): /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include/
  Processing folder (recursive): /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/gcc/x86_64-conda-linux-gnu/10.3.0/include-fixed/
  Processing folder (recursive): /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/include/
Failed to spawn IntelliSense process:
Shutting down IntelliSense server: /workspaces/pfs/drp_stella/src/FiberTraceSet.cc
cpptools/getFoldingRanges: /workspaces/pfs/drp_stella/src/FiberTraceSet.cc (id: 3)
cpptools/getSemanticTokens: /workspaces/pfs/drp_stella/src/FiberTraceSet.cc (id: 4)
Checking for syntax errors: /workspaces/pfs/drp_stella/src/FiberTraceSet.cc
Failed to spawn IntelliSense process:
Shutting down IntelliSense server: /workspaces/pfs/drp_stella/src/FiberTraceSet.cc
  Processing folder (recursive): /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/x86_64-conda-linux-gnu/sysroot/usr/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/afw/g31dc256b76+308b63831b/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/astshim/gb04e0da5b2+ce83317b7e/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/base/gc57f72a14a+ce83317b7e/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/cpputils/g1ffd6c4751+50c04d9c73/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/daf_base/gcc072456be+b6f9ac6643/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/eigen/g04a8d4365e/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/geom/g491c310abf+35ead245fa/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/log/gd27fc4addd+b6f9ac6643/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/meas_algorithms/gf36ae6ace1+1d7ae49801/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/meas_base/g0129b61b94+2131942eb1/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/pex_config/g9aff22451c+c55107240f/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/pex_exceptions/g958458c828+fc3c6d1a77/include/
  Processing folder (recursive): /opt/pfs/stack/miniconda3-py38_4.9.2-3.0.0/Linux64/sphgeom/g0f31348038+ce83317b7e/include/
  Processing folder (recursive): /opt/pfs/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/include/
```
* After a while, I get a "cpptools" log output:
```
[Error - 11:01:40 PM] Connection to server got closed. Server will not be restarted.
```

**Comment by Colengms** [comment]
Hi @PaulPrice .  So far, this is the first report of this behavior.  There would seem to be something particular about your environment, or we would likely be getting a flood of such reports.  Could you provide your devcontainer files?

Could you also try using our latest insiders build, 1.10.0.  It's available in the VS Code Marketplace UI, using the `Switch to Pre-Release` option.

If you'd like to work around the issue by doing the chmods yourself, you would need to chmod the other executables in the bin directory, including `cpptools-srv`.  Though, given the unexpected behavior, there may other issues lurking.

**Comment by PaulPrice** [comment]
It seems likely that this is peculiar to my environment, but I hope that if we can get to the bottom of it we can make the extension more robust for everyone. In any case, I appreciate your help on figuring this out, as my experience with this extension is that it *drastically* improves my C++ coding efficiency.

Here are the relevant files:
* [Dockerfile.pipe2d](https://github.com/Subaru-PFS/pfs_pipe2d/blob/master/docker/Dockerfile.pipe2d) (based off another [Dockerfile in the same directory](https://github.com/Subaru-PFS/pfs_pipe2d/blob/master/docker/Dockerfile.lsst)).
* [devcontainer.json](https://gist.github.com/PaulPrice/9072555b23c22af38a9f1b56295da454)
* [devcontainer.env](https://gist.github.com/PaulPrice/cbabbeabce0e2f1e5086565b0da74a22)
* [c_cpp_properties.json](https://gist.github.com/PaulPrice/8914b3ba6589a39d0bab8c0a5b4aaad2)

Same result with the pre-release (1.10.0). After the install, I found the following in the "Log (Window)", which might be useful:
```
[2022-05-03 09:51:32.771] [renderer1] [info] Ignoring configured affinity for 'pub.name' because the extension is unknown or cannot execute.
[2022-05-03 09:51:44.851] [renderer1] [info] Ignoring configured affinity for 'pub.name' because the extension is unknown or cannot execute.
[2022-05-03 09:51:45.040] [renderer1] [info] Invoking resolveAuthority(dev-container)
[2022-05-03 09:51:59.873] [renderer1] [info] resolveAuthority(dev-container) returned '127.0.0.1:35821' after 14833 ms
[2022-05-03 09:52:02.626] [renderer1] [info] Ignoring configured affinity for 'pub.name' because the extension is unknown or cannot execute.
[2022-05-03 09:52:02.654] [renderer1] [warning] [twxs.cmake]: Cannot register 'cmake.cmakePath'. This property is already registered.
[2022-05-03 09:53:21.132] [renderer1] [info] Ignoring configured affinity for 'pub.name' because the extension is unknown or cannot execute.
[2022-05-03 09:53:24.554] [renderer1] [error] spawn /home/pfs/.vscode-server/extensions/ms-vscode.cpptools-1.10.0-linux-x64/bin/cpptools EACCES: Error: spawn /home/pfs/.vscode-server/extensions/ms-vscode.cpptools-1.10.0-linux-x64/bin/cpptools EACCES
	at Process.ChildProcess._handle.onexit (node:internal/child_process:282:19)
	at onErrorNT (node:internal/child_process:477:16)
	at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

**Comment by PaulPrice** [comment]
Another potential clue: 1.7.1 works fine. 1.8.x and 1.9.7 are apparently not compatible with VSCode 1.66.2, otherwise I would try them.

**Comment by PaulPrice** [comment]
I suspect this is caused somehow by my mounting the container home directory. My `.devcontainer.json` has:
```
    "mounts": [
        "source=${localWorkspaceFolder}/.devcontainer/home,target=/home/pfs,type=bind,consistency=delegated",
        "source=${localEnv:HOME}/LSST,target=/home/pfs/LSST,type=bind,consistency=delegated",
        "source=${localEnv:HOME}/pfs,target=/home/pfs/pfs,type=bind,consistency=delegated",
    ],
```

If I comment out those mounts, then everything works nicely. I originally put them there so that I could configure my environment through a shell startup script. But the shell startup script kept timing out (I got errors about `userEnvProbe` taking too long: the timeout seems to be set to a couple of seconds), so I changed to specifying the environment variables directly to docker via the `--env-file` argument in the `runArgs`. Then I don't need the mounts and everything works.

I'm happy to close this at this point, unless you're interested in further feedback on corner cases with the mounts and/or `userEnvProbe` timeouts.
